plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -60,23 +80,53 @@ protected void setup(Binder binder)
         newOptionalBinder(binder, Key.get(UserNameProvider.class, ForHiveMetastore.class))
                 .setDefault()
                 .toInstance(SIMPLE_USER_NAME_PROVIDER);
-
         binder.bind(Key.get(boolean.class, AllowHiveTableRename.class)).toInstance(true);
 
-        install(new ThriftMetastoreAuthenticationModule());
+        validateThriftHttpConfiguration();
     }
 
-    @Provides
-    @Singleton
-    @ThriftHiveWriteStatisticsExecutor
-    public ExecutorService createWriteStatisticsExecutor(ThriftMetastoreConfig hiveConfig)
+    private void validateThriftHttpConfiguration()
```

Comment:
Can we move this method to within `hasHttp(s)Metastore` and inline the validation?